lxd/container.go

```diff
@@ -229,14 +229,28 @@ func isRootDiskDevice(device types.Device) bool {
 	return false
 }
 
-func containerGetRootDiskDevice(devices types.Devices) (string, types.Device) {
-	for devName, dev := range devices {
-		if isRootDiskDevice(dev) {
-			return devName, dev
+func containerGetRootDiskDevice(devices types.Devices) (string, types.Device, error) {
+	var devName string
+	var dev types.Device
+
+	count := 0
+	for n, d := range devices {
+		if isRootDiskDevice(d) {
+			count += 1
+			devName = n
+			dev = d
 		}
```

Comment:
This is pedantry but performing the check for `if count > 1` directly after the `isRootDiskDevice` check in the loop and then return immediately might be useful in a container with a lot of devices. But it's not a merge blocker obvs.